source/common/config/xds_mux/grpc_mux_impl.cc

```diff
@@ -0,0 +1,383 @@
+#include "source/common/config/xds_mux/grpc_mux_impl.h"
+
+#include "envoy/service/discovery/v3/discovery.pb.h"
+
+#include "source/common/common/assert.h"
+#include "source/common/common/backoff_strategy.h"
+#include "source/common/config/decoded_resource_impl.h"
+#include "source/common/config/utility.h"
+#include "source/common/config/version_converter.h"
+#include "source/common/config/xds_context_params.h"
+#include "source/common/config/xds_resource.h"
+#include "source/common/memory/utils.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Config {
+namespace XdsMux {
+
+template <class S, class F, class RQ, class RS>
```

Comment:
Is it explained somewhere what each of these class template parameters are?

I know we discussed this before and it seemed difficult after investigation, but anything we can do to reduce the number of template variables here will aid readability/debugability.

Reply:
Will add comments; I tried reducing the number of template args, but it's not really possible due to lack of a common interface between Delta and Sotw discovery requests and responses. The original implementation was using void pointers to work around this, but the consensus was that strong typing was preferable.

Reply:
done